video_xblock/video_xblock.py

```diff
@@ -884,6 +989,69 @@ def authenticate_video_api(self, token=''):
         return auth_data, error_message
 
     @XBlock.json_handler
+    def get_transcripts_3playmedia_api_handler(self, data, suffix=''):  # pylint: disable=unused-argument
+        """
+        Xblock handler to authenticate to a video platform's API. Called by JavaScript of `studio_view`.
+
+        Arguments:
+            data (dict): Data from frontend, necessary for authentication (tokens, account id, etc).
+            suffix (str): Slug used for routing.
+        Returns:
+            response (dict): Status messages key-value pairs.
+        """
+        domain = 'http://static.3playmedia.com/'
+        video_id = self.get_player().media_id(self.href)
+        error_message = ''
+        _transcripts = []
+        apikey = data.get('api_key', self.playmedia_apikey) or ''
+        file_id = data.get('file_id', '')
+
+        transcripts_3playmedia = requests.get(
+            '{domain}files/{file_id}/translations?apikey={api_key}'.format(
+                domain=domain, file_id=file_id, api_key=apikey
+            )
+        ).json()
+        errors = isinstance(transcripts_3playmedia, dict) and transcripts_3playmedia.get('errors')
+        if errors:
+            return {'error_message': u'\n'.join(errors.values())}
+
+        transcript_original = requests.get(
+            '{domain}files/{file_id}/transcript.vtt?apikey={api_key}'.format(
+                domain=domain, file_id=file_id, api_key=apikey
+            )
+        ).text
+        _transcripts.append(
+            self.convert_3playmedia_caps_to_vtt(
+                caps=transcript_original, video_id=video_id
+            )
+        )
+        for transcript in transcripts_3playmedia:
+            tid = transcript.get('id', '')
+            # File name format is <language label>_captions_video_<video_id>, e.g.
+            # "English_captions_video_456g68"
+            sub_unicode = requests.get(
+                '{domain}files/{file_id}/translations/{tid}/captions.vtt?apikey={api_key}'.format(
+                    domain=domain, file_id=file_id, api_key=apikey, tid=tid
+                )
+            ).text
+            _transcripts.append(
+                self.convert_3playmedia_caps_to_vtt(
+                    caps=sub_unicode,
+                    video_id=video_id,
+                    lang=transcript.get('target_language_iso_639_1_code', ''),
+                    lang_label=transcript.get('target_language_name', '')
+                )
+            )
+
+        if error_message:
+            response = {'error_message': error_message}
+        else:
+            success_message = 'Successfully authenticated to the video platform.'
```

Comment:
Add i18n and change the text to something like:
`'Successfully fetched transcripts from 3playMedia. Please check transcripts list above.'`